src/Ads/AssetSuggestionsService.php

```diff
@@ -17,13 +19,181 @@
  */
 class AssetSuggestionsService implements Service {
 
+	/**
+	 * Default maximum marketing images.
+	 */
+	protected const DEFAULT_MAXIMUM_MARKETING_IMAGES = 20;
+
 	/**
 	 * AssetSuggestionsService constructor.
 	 *
-	 * @param WP $wp
+	 * @param WP $wp WP Proxy.
+	 * @param WC $wc WC Proxy.
 	 */
-	public function __construct( WP $wp ) {
+	public function __construct( WP $wp, WC $wc ) {
 		$this->wp = $wp;
+		$this->wc = $wc;
+	}
+
+	/**
+	 * Get assets from specific post or term.
+	 *
+	 * @param int    $id Post or Term ID.
+	 * @param string $type Only possible values are post or term.
+	 */
+	public function get_assets_suggestions( int $id, string $type ): array {
+		// TODO: Fetch assets from others campaigns and merge the result with the WP Assets.
+		return $this->get_wp_assets( $id, $type );
+	}
+
+	/**
+	 * Get assets from specific post or term.
+	 *
+	 * @param int    $id Post or Term ID.
+	 * @param string $type Only possible values are post or term.
+	 *
+	 * @return array All assets available for specific term or post.
+	 */
+	protected function get_wp_assets( int $id, string $type ): array {
+		if ( $type === 'post' ) {
+			return $this->get_post_assets( $id );
+		}
+
+		return [];
+
+	}
+
+	/**
+	 * Get assets from specific post.
+	 *
+	 * @param int $id Post ID.
+	 *
+	 * @return array All assets for specific post.
+	 * @throws Exception If the Post ID is invalid.
+	 */
+	protected function get_post_assets( int $id ): array {
+		$post = $this->wp->get_post( $id );
+
+		if ( ! $post ) {
+			throw new Exception(
+				/* translators: 1: is a integer representing an unknown Post ID */
+				sprintf( __( 'Invalid Post ID %1$d', 'google-listings-and-ads' ), $id )
+			);
+		}
+
+		$attachments_ids = $this->get_post_attachments(
+			[
+				'post_parent' => $id,
+			]
+		);
+
+		if ( $id === wc_get_page_id( 'shop' ) ) {
+			$attachments_ids = array_merge( $attachments_ids, $this->get_shop_attachments() );
+		}
+
+		if ( $post->post_type === 'product' || $post->post_type === 'product_variation' ) {
+			$product         = $this->wc->maybe_get_product( $id );
+			$attachments_ids = array_merge( $attachments_ids, $product->get_gallery_image_ids(), [ $product->get_image_id() ] );
+		}
+
+		$gallery_images_urls = get_post_gallery_images( $id );
+		$marketing_images    = array_merge( $this->get_url_attachments_by_ids( $attachments_ids ), $gallery_images_urls );
+		$marketing_images    = array_slice( $marketing_images, 0, self::DEFAULT_MAXIMUM_MARKETING_IMAGES );
+		$long_headline       = get_bloginfo( 'name' ) . ': ' . $post->post_title;
+
+		return [
+			'headline'                => [ $post->post_title ],
+			'long_headline'           => [ $long_headline ],
+			'description'             => $this->remove_empty_values( [ $post->post_excerpt, get_bloginfo( 'description' ) ] ),
```

Comment:
❔From the PR description:
> The Headlines (1 & 2) are grouped as one headline
The Descriptions (4 & 5) are grouped as one description

I can see there's only one `description` in the response, but still has two headline: `headline` and `long_headline`. Not sure if I understand it correctly, should it only be just `headline`?

Reply:
> headline and long_headline. Not sure if I understand it correctly, should it only be just headline?

Google Ads differentiate between, [HEADLINE](https://github.com/googleads/google-ads-php/blob/151bab9bd37d46543bbeb2c3b0caedf91cb91b6b/src/Google/Ads/GoogleAds/V11/Enums/AssetFieldTypeEnum/AssetFieldType.php#L177) and [LONG_HEADLINE](https://github.com/googleads/google-ads-php/blob/151bab9bd37d46543bbeb2c3b0caedf91cb91b6b/src/Google/Ads/GoogleAds/V11/Enums/AssetFieldTypeEnum/AssetFieldType.php#L193) thats why I kept both fields. For descriptions, Google Ads contains only one type.